migration/src/main/java/uk/ac/bbsrc/tgac/miso/migration/destination/ValueTypeLookup.java

```diff
@@ -0,0 +1,545 @@
+    setSamplePurposes(misoServiceManager.getSamplePurposeDao().getSamplePurpose());
+    setLabs(misoServiceManager.getLabDao().getLabs());
+    setTissueOrigins(misoServiceManager.getTissueOriginDao().getTissueOrigin());
+    setLibrarySelections(misoServiceManager.getLibraryDao().listAllLibrarySelectionTypes());
+    setLibraryTypes(misoServiceManager.getLibraryDao().listAllLibraryTypes());
+    setLibraryDesigns(misoServiceManager.getLibraryDesignDao().getLibraryDesigns());
+    setTagBarcodes(misoServiceManager.getTagBarcodeDao().listAllTagBarcodes());
+  }
+  
+  private void setSampleClasses(Collection<SampleClass> sampleClasses) {
+    Map<Long, SampleClass> mapById = new HashMap<>();
+    Map<String, SampleClass> mapByAlias = new HashMap<>();
+    for (SampleClass sampleClass : sampleClasses) {
+      if (mapByAlias.containsKey(sampleClass.getAlias())) {
+        throw new IllegalStateException("Duplicate SampleClass alias: " + sampleClass.getAlias());
+      }
+      mapByAlias.put(sampleClass.getAlias(), sampleClass);
+      mapById.put(sampleClass.getId(), sampleClass);
+    }
+    this.sampleClassByAlias = mapByAlias;
+    this.sampleClassById = mapById;
+  }
+  
+  private void setTissueTypes(Collection<TissueType> tissueTypes) {
+    Map<Long, TissueType> mapById = new HashMap<>();
+    Map<String, TissueType> mapByAlias = new HashMap<>();
+    for (TissueType tt : tissueTypes) {
+      if (mapByAlias.containsKey(tt.getAlias())) {
+        throw new IllegalStateException("Duplicate TissueType alias: " + tt.getAlias());
+      }
+      mapByAlias.put(tt.getAlias(), tt);
+      mapById.put(tt.getId(), tt);
+    }
+    this.tissueTypeById = mapById;
+    this.tissueTypeByAlias = mapByAlias;
+  }
+  
+  private void setTissueMaterials(Collection<TissueMaterial> tissueMaterials) {
+    Map<Long, TissueMaterial> mapById = new HashMap<>();
+    Map<String, TissueMaterial> mapByAlias = new HashMap<>();
+    for (TissueMaterial tm : tissueMaterials) {
+      if (mapByAlias.containsKey(tm.getAlias())) {
+        throw new IllegalStateException("Duplicate TissueMaterial alias: " + tm.getAlias());
+      }
+      mapByAlias.put(tm.getAlias(), tm);
+      mapById.put(tm.getId(), tm);
+    }
+    this.tissueMaterialById = mapById;
+    this.tissueMaterialByAlias = mapByAlias;
+  }
+  
+  private void setKits(Collection<KitDescriptor> kits) {
+    Map<Long, KitDescriptor> mapById = new HashMap<>();
+    Map<String, KitDescriptor> mapByName = new HashMap<>();
+    for (KitDescriptor kit : kits) {
+      if (mapByName.containsKey(kit.getName())) {
+        throw new IllegalStateException("Duplicate KitDescriptor name: " + kit.getName());
+      }
+      mapByName.put(kit.getName(), kit);
+      mapById.put(kit.getId(), kit);
+    }
+    this.kitById = mapById;
+    this.kitByName = mapByName;
+  }
+  
+  private void setSamplePurposes(Collection<SamplePurpose> samplePurposes) {
+    Map<Long, SamplePurpose> mapById = new HashMap<>();
```

Comment:
Is it worth creating an implementation of map that throws a duplicate key on `put` to avoid doing this so many times?